sos/cleaner/mappings/ipv6_map.py

```diff
@@ -0,0 +1,264 @@
+        used to create the instance, as well as on an optional ``obfuscation``
+        which if set, will serve as the obfuscated_network address.
+
+        :param addr:        The *un*obfuscated network to be handled
+        :type addr:         ``ipaddress.IPv6Network``
+
+        :param obfuscation: An optional pre-determined string representation of
+                            the obfuscated network address
+        :type obfuscation:  ``str``
+        """
+        if not isinstance(addr, ipaddress.IPv6Network):
+            raise Exception('Invalid network: not an IPv6Network object')
+        self.addr = addr
+        self.prefix = addr.prefixlen
+        self.network_addr = addr.network_address.compressed
+        self.hosts = {}
+        self._first_hexes = ['534f']
+        if not obfuscation:
+            self._obfuscated_network = self._obfuscate_network_address()
+        else:
+            if not isinstance(obfuscation, str):
+                raise TypeError(f"Pre-determined obfuscated network address "
+                                f"must be str, not {type(obfuscation)}")
+            self._obfuscated_network = obfuscation.split('/')[0]
+
+    @property
+    def obfuscated_address(self):
+        return f"{self._obfuscated_network}/{self.prefix}"
+
+    @property
+    def original_address(self):
+        return self.addr.compressed
+
+    def _obfuscate_network_address(self):
+        """Generate the obfuscated pair for the network address. This is
+        determined based on the netmask of the network this class was built
+        on top of.
+        """
+        if self.addr.is_global:
+            return self._obfuscate_global_address()
+        elif self.addr.is_link_local:
+            # link-local addresses are always fe80::/64. This is not sensitive
+            # in itself, and retaining the information that an address is a
+            # link-local address is important for problem analysis, so don't
+            # obfuscate this network information.
+            return self.network_addr
+        elif self.addr.is_private:
+            return self._obfuscate_private_address()
+
+    def _obfuscate_global_address(self):
+        """Global unicast addresses have a 48-bit global routing prefix and a
+        16-bit subnet. We set the global routing prefix to a static
+        sos-specific identifier that could never be seen in the wild,
+        '534f:'
+
+        We then randomize the subnet hextet.
+        """
+        _hextets = self.network_addr.split(':')[1:]
+        _ob_hex = ['534f']
+        if all(not c for c in _hextets):
+            # we have only a single defined hextet, e.g. ff00::/64, so we need
+            # to not use the standard first-hex identifier or we'll overlap
+            # every similar address obfuscation
+            _ob_hex = f"53{random_hex(2)}"
+            while _ob_hex in self._first_hexes:
+                # prevent duplicates
+                _ob_hex = f"53{random_hex(2)}"
```

Comment:
We are limited here by 256 options only, and I can imagine some systems referring to more uniq global addresses - in that case, cleaner would loop here forever. Not sure how realistic the scenario / problem is, though.

Reply:
Good idea, we wouldn't be obfuscating anything else with that prefix as the first bits in the address in any condition, so makes sense to increment from there.

Reply:
Added incrementing from 53 on up for every 256 addresses generated like this.